docs/authoring-workspaces.md

```diff
@@ -0,0 +1,107 @@
+
+<!-- markdownlint-disable-next-line MD013 -->
+Azure TRE workspaces are [Porter](https://porter.sh/) bundles that in turn are based on [Cloud Native Application Bundles (CNAB)](https://cnab.io/). Workspace authors are free to choose the technology stack for provisioning resources (e.g., ARM templates, Terraform etc.), but the Azure TRE framework sets certain requirements for the bundle manifests, which specify the credentials, input and output parameters, deployment actions among other things. The document describes those requirements.
+
+**Use [the vanilla workspace bundle](../workspaces/vanilla/README.md) and [others](../workspaces/) as reference** or as the basis for your new bundle. To start from scratch follow the Porter [Quickstart Guide](https://porter.sh/quickstart/) ([`porter create` CLI command](https://porter.sh/cli/porter_create/) will generate a new bundle in the current directory).
+
+## Prerequisites
+
+* [Docker installed](https://docs.docker.com/get-docker/)
+* [Porter installed](https://porter.sh/install)
+* Azure TRE instance deployed to test against
+
+## Workspace bundle manifest
+
+The manifest of a workspace bundle is the `porter.yaml` file (see [Author Bundles in Porter documentation](https://porter.sh/author-bundles/)). This section describes the mandatory credentials, input and output parameters of a TRE workspace bundle.
+
+### Credentials
+
+A workspace bundle requires the following [credentials](https://porter.sh/author-bundles/#credentials) to provision resources in Azure:
+
+* [Azure tenant ID](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-how-to-find-tenant)
+* Azure subscription ID
+* The client ID of a [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) with privileges to provision resources
+* The client secret (password) of a service principal
+
+The credentials are provided as environment variables by the deployment runner. The bundle author must use the following environment variable names:
+
+* `AZURE_TENANT_ID`
+* `AZURE_SUBSCRIPTION_ID`
+* `AZURE_SP_CLIENT_ID`
+* `AZURE_SP_PASSWORD`
+
+The names of the Porter credentials (`name` field in `porter.yaml`) can be freely chosen by the author.
+
+Example:
+
+```yaml
+credentials:
+- name: azure_tenant_id
+  env: AZURE_TENANT_ID
+- name: azure_subscription_id
+  env: AZURE_SUBSCRIPTION_ID
+- name: azure_service_principal_client_id
+  env: AZURE_SP_CLIENT_ID
+- name: azure_service_principal_password
+  env: AZURE_SP_PASSWORD
+```
+
+### Parameters
+
+This section describes the mandatory [(input) parameters](https://porter.sh/author-bundles/#parameters) of a workspace bundle manifest.
+
+| Parameter | Type | Description | Example value |
+| --------- | ---- | ----------- | ------------- |
+| `core_id` | string | The ID of the parent TRE instance. | `mytre-dev-3142` |
+| `workspace_id` | string | Unique 4-digit workspace ID. | `2718` |
```

Comment:
Why 4 digits? We do have a unique name for a Workspace, but i don't think we have an id currently. Right @TessFerrandez ?

Reply:
I can at least tell you that it shouldn't be much longer since it's used in Azure resource name(s), combined with the core ID, which do have a max length.

Reply:
Ahh- it is used to make the RG name unique. I assume this will be generated by the API, which should also make sure that the RG name is not too long